src/page/login/login.tsx

```diff
@@ -17,19 +19,185 @@ import LinearGradient from 'react-native-linear-gradient';
 import {
   updateAccount,
   updatePassword,
+  updateLoginstatus,
   fetchData,
 } from '../../store/login/action';
+import FadeIn from '../../components/fadeIn';
+import PopUpTips from '../../components/popupTips';
 
 const Login = () => {
   const navitation = useNavigation();
   const state = useSelector(state => state.login);
+  const [valAccountMss, setValAccountMss] = useState('');
+  const [valPasswordMss, setValPasswordMss] = useState('');
+  const [popupIndex, setPopupIndex] = useState(-1);
   const dispatch = useDispatch();
+  const durationTime = useRef(300).current;
+
+  let commaDisOne = useRef(new Animated.Value(26)).current;
```

Comment:
It's cool that you figured out with animation  API but your animation is too complicated here. You may use instead [LayoutAnimation](https://medium.com/@Jpoliachik/react-native-s-layoutanimation-is-awesome-4a4d317afd3e) to change width of the button and `react-native-lottie` to have beautiful loading indicator. Usually, we need just one or two animated values (no more) and the just interpolate it - to animate different styles.

```
const animatedValue = useMemo(() => new Animate.Value(0), [])

const opacity = animatedValue.interpolate({})
const translateX = animatedValue.interpolate({})
const width = animatedValue.interpolate({})

....
```

and etc.